.github/actions/triggerWorkflowAndWait/triggerWorkflowAndWait.js

```diff
@@ -0,0 +1,138 @@
+                                newWorkflowRunID = lastWorkflowRunID;
+                                hasNewWorkflowStarted = newWorkflowRunID !== previousWorkflowRunID;
+
+                                if (!hasNewWorkflowStarted) {
+                                    waitTimer += POLL_RATE;
+                                    if (waitTimer < NEW_WORKFLOW_TIMEOUT) {
+                                        // eslint-disable-next-line max-len
+                                        console.log(`After ${waitTimer / 1000} seconds, there's still no new ${workflow} workflow run 🙁`);
+                                    } else {
+                                        // eslint-disable-next-line max-len
+                                        const err = new Error(`After ${NEW_WORKFLOW_TIMEOUT} seconds, the ${workflow} workflow did not start.`);
+                                        console.error(err);
+                                        core.setFailed(err);
+                                        process.exit(1);
+                                    }
+                                } else {
+                                    console.log(`\n🚀 New ${workflow} run with ID ${newWorkflowRunID} has started`);
+                                }
+                            })
+                            .catch((err) => {
+                                console.warn('Failed to fetch latest workflow run.', err);
+                            });
+                    },
+                    POLL_RATE,
+                ),
+            );
+        })
+
+        // Wait for the new workflow run to finish
+        .then(() => promiseWhile(
+            () => !workflowCompleted,
+            _.throttle(
+                () => {
+                    console.log(`\n⏳ Waiting for workflow run ${newWorkflowRunID} to finish...`);
+                    return octokit.actions.getWorkflowRun({
+                        owner: GithubUtils.GITHUB_OWNER,
+                        repo: GithubUtils.EXPENSIFY_CASH_REPO,
+                        run_id: newWorkflowRunID,
+                    })
+                        .then(({data}) => {
+                            workflowCompleted = data.status === 'completed' && data.conclusion !== null;
+                            if (workflowCompleted) {
+                                if (data.conclusion === 'success') {
+                                    console.log(`\n🎉 ${workflow} run ${newWorkflowRunID} completed successfully! 🎉`);
+                                } else {
+                                    // eslint-disable-next-line max-len
+                                    const err = new Error(`🙅‍ ${workflow} run ${newWorkflowRunID} finished with conclusion ${data.conclusion}`);
+                                    console.error(err.message);
+                                    core.setFailed(err);
+                                    process.exit(1);
```

Comment:
This is the same as the `catch` implementation above, I think it'd be cleaner to move the catch to the end of the chain and handle this there

Reply:
I am open to ideas if you feel this needs to be cleaned/DRY'd up, but these two blocks do have different error messages, which is why they are separate right now.

Reply:
Gotcha, https://github.com/Expensify/Expensify.cash/pull/2495/files#r629728390 makes sense so I think this is fine